kelpy/deployment.py

```diff
@@ -40,6 +41,28 @@ def get(client, name, namespace="default"):
     return response
 
 
+def update(client, name, body, namespace="default", timeout=100):
+    try:
+        response = client.patch_namespaced_deployment(name, namespace, body)
+    except ApiException as e:
+        if e.reason == "Conflict":
+            return False
+        raise e
+
+    w = watch.Watch()
+    for event in w.stream(
+        client.list_deployment_for_all_namespaces, timeout_seconds=timeout
+    ):
+        if (
+            event["type"] == "ADDED"
+            and event["object"].metadata.name == response.metadata.name
+            and event["object"].status.observed_generation == 2
```

Comment:
I don't fully understand this field, but I'm imagining that if you were to update a deployment multiple times, this value will increment each time, I think it would either need to be looked up and then we check if it's incremented, or confirm that it's actually required.

Reply:
Thanks for the comments. I have updated the code So that it will wait until the replicas are available with a timeout value. Please review